### PR TITLE
[hapi__joi] Fix parameter type used in type def for ValidationErrorFunction

### PR DESCRIPTION
**Pull request by SimchaYork**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: 
https://github.com/hapijs/joi/blob/master/lib/base.js#L634
https://github.com/hapijs/joi/blob/master/lib/validator.js#L460

### CODE REVIEW

File: types/hapi__joi/index.d.ts [hunk 1]
```diff
@@ -190,6 +194,17 @@ declare namespace Joi {
         warnings?: boolean;
     }
 
+    interface LanguageMessageTemplate {
+        source: string;
+        rendered: string;
+        _template: null | any[];
```
Comment (SimonSchick) [inline types/hapi__joi/index.d.ts]: `_` usually means internal and should not be exposed in typings.

Comment (SimchaWood) [reply types/hapi__joi/index.d.ts]: _template and _settings have been removed from interface.

**Review by SimonSchick — CHANGES_REQUESTED** [review]
Forgot to ask, can you please add a test that demonstrates/ensures your changes?

**Review by SimonSchick — CHANGES_REQUESTED** [review]
Please revert the formatting changes, it makes things hard to review.

**Review by SimonSchick — APPROVED** [review]

**Pull request merged**